src/thunderbots/software/ai/hl/stp/evaluation/calc_best_shot.h

```diff
@@ -9,82 +8,133 @@
-     * goal that can be seen from a point.
-     *
-     * @param f-field is needed to calculate the location of the goalposts based on field
-     * length and goal width
-     *
-     * @param obstacles-obstacles is a list of all the obstacles in the way between the
-     * point and the net
+     * Finds the best shot on the given goal, and returns the best target to shoot at and
+     * the largest open angle interval for the shot (this is the total angle between the
+     * obstacles on either side of the shot vector).
+     *
+     * @param goal_post_neg The goal post of the net with the negative y-coordinate
+     * @param goal_post_posThe goal post of the net with the positive y-coordinate
+     * @param p The point that the shot will be taken from
+     * @param obstacles The locations of any obstacles on the field that may obstruct the
+     * shot. These are treated as circular obstacles, and are usually used to represent
+     * robots on the field.
+     * @param radius The radius of the given obstacles
+     *
+     * @return the best target to shoot at and the largest open angle interval for the
+     * shot (this is the total angle between the obstacles on either side of the shot
+     * vector). If no shot is possible, the returned angle will be 0
+     */
+    std::pair<Point, Angle> calcBestShotOnGoal(const Point &goal_post_neg,
+                                               const Point &goal_post_pos, const Point &p,
+                                               const std::vector<Point> &obstacles,
+                                               double radius);
+
+    /**
+     * Finds the best shot on the specified goal, and returns the best target to shoot at
+     * and the largest open angle interval for the shot (this is the total angle between
+     * the obstacles on either side of the shot vector).
+     *
+     * This function will treat all robots on the field as obstacles, except for the
+     * robots_to_ignore
+     *
+     * @param world The world state
+     * @param point The point that the shot will be taken from
+     * @param radius The radius for the robot obstacles
+     * @param robots_to_ignore The robots to ignore
+     * @param shoot_at_enemy_goal Whether or not the shot is at the enemy goal. If true,
+     * the shot will be treated as a shot at the enemy goal, and if false, the shot will
+     * be treated as a shot at the friendly goal
+     *
+     * @return the best target to shoot at and the largest open angle interval for the
+     * shot (this is the total angle between the obstacles on either side of the shot
+     * vector).
+     */
+    std::pair<Point, Angle> calcBestShotOnGoal(const World &world, const Point &point,
+                                               double radius,
+                                               const std::vector<Robot> &robots_to_ignore,
+                                               bool shoot_at_enemy_goal);
```

Comment:
I'm a bit confused, the `shoot_at_enemy_goal` doesn't appear anywhere else in this PR? Regardless, this seems a kinda icky way to do this, I think just have the two separate functions is probably a better call.